### ISSUE TITLE
Data Augmentation Sampling Strategy

### ISSUE THREAD

**Issue by wscds** [Hlings/AcroFOD#2]
The paper mentioned that the augmentation sample data from two domain，but the data augmentation in the code seems to only sample data from one domain，is the source image [A, B] means source and target image？But if the target image is few, the few shot image is difficult to be selected。Would it be okay to do so?

And another question is what is the role of the variable bs_add(or image_add).

**Comment by gaoypeng** [comment]
Q1: Yes. [A, B] means merging a large number of source images A and limited target images B together.

Q2: Is such an operation ok if target images are few? Yes. To be honest, I was also surprised by this discovery. I tried some further exps eg. repeating B more times like [A, B, B, B] to introduce more target images. The results are similar. You may also improve this operation during training in other ways :). Hope you get better results.

Q3: What's the role of bs_add? In top-k selection, some data will be filtered, the amount is around total_bs*(1-k). To keep the total training samples remain the same for backpropagation (loss scale), I copy the first bs_add samples in the selected samples. 

Hope the above reply is helpful to you :). I will also write more detailed instructions following.
